src/components/UI/ModalWindows/SubmitProposalModal/SubmitProposalModal.tsx

```diff
@@ -13,69 +13,94 @@ import { MAX_LENGTH_OF_COVER_LETTER } from "utils/consts/numberConsts";
 import { ProposalType } from "utils/enums";
 import { useGetOwnProfileQuery } from "services/profileInfo/profileInfoAPI";
 
+import { InboxOutlined } from "@ant-design/icons";
+import { RcFile } from "antd/lib/upload";
+
 const { Text } = Typography;
 
 interface ISubmitProposalForm extends FieldValues {
   bid: number;
-  exampleRequired: string;
+  message: string;
+  file: RcFile | undefined;
 }
 
 interface ISubmitProposalModalProps {
   idJobPost: number;
   visible: boolean;
   closeModal: () => void;
   clientBudget: number;
+  refetch: () => void;
 }
 
 function SubmitProposalModal({
   idJobPost,
   visible,
   closeModal,
   clientBudget,
+  refetch,
 }: ISubmitProposalModalProps) {
   const { t } = useTranslation();
 
+  const [selectedFile, setSelectedFile] = useState<RcFile | undefined>(
+    undefined
+  );
+
   const { data: freelancer } = useGetOwnProfileQuery();
   const { rate, id } = freelancer!;
 
-  const { control, handleSubmit, reset } = useForm<ISubmitProposalForm>({
-    resolver: yupResolver(submitProposalSchema),
-    defaultValues: {
-      bid: +rate,
-    },
-  });
+  const { control, setValue, handleSubmit, reset } =
+    useForm<ISubmitProposalForm>({
+      resolver: yupResolver(submitProposalSchema),
+      defaultValues: {
+        bid: +rate,
+      },
+    });
 
   const [runSendProposalMutation, { isError, isLoading, isSuccess }] =
     useSendProposalMutation();
 
-  const onSubmit: SubmitHandler<ISubmitProposalForm> = async ({
-    bid,
-    coverLetter,
-  }) => {
-    if (freelancer)
-      await runSendProposalMutation({
-        bid,
-        message: coverLetter,
-        idJobPost,
-        idFreelancer: id,
-        type: ProposalType.PROPOSAL,
-      });
-    closeModal();
-
-    reset();
-  };
-
   useEffect(() => {
     if (!isLoading && isError) {
       toast.error(t("ServerErrors.FETCH_ERROR"));
       return;
     }
     if (!isLoading && isSuccess) {
       toast.success(t("SearchWork.success"));
-      reset();
+      setSelectedFile(undefined);
+      refetch();
     }
+
+    setValue("file", selectedFile!);
+
     // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [isLoading]);
+  }, [isLoading, selectedFile]);
+
+  const onSubmit: SubmitHandler<ISubmitProposalForm> = async (data) => {
+    const formData = new FormData();
+
+    const reqObject = {
+      ...data,
+      idJobPost,
+      idFreelancer: id,
+      type: ProposalType.PROPOSAL,
+    };
+
+    // eslint-disable-next-line array-callback-return
+    Object.entries(reqObject).map(([key, value]) => {
+      if (Array.isArray(value)) {
+        value.map((element, index) =>
+          formData.append(`${key}[${index}]`, `${element}`)
+        );
+      } else {
+        formData.append(key, value instanceof File ? value : `${value}`);
+      }
+    });
+
+    if (freelancer) await runSendProposalMutation(formData);
+
+    closeModal();
+    reset();
+  };
```

Comment:
what if you move all this logic into the custom hook? you will isolate all the logic